Owlcat/bin/flag-ms.py

```diff
@@ -585,3 +608,49 @@ def parse_subset_options(options):
             traceback.print_exc()
             error("Error exporting flags to %s" % options.export)
         print("Flags exported OK.")
+
+    if options.save:
+        flagvers = f"{msname}.flagversions/flags.{options.save}"
+        if Owlcat.tableexists(flagvers) and not options.force:
+            error(f"Flagversion '{options.save}' already exists. Please run with --force to confirm overwrite.")
+        try:
+            ms = get_ms(readonly=False)
+            nr = ms.nrows()
+            flag_row = ms.getcol("FLAG_ROW")
+            flag_col = ms.getcol("FLAG")
+            fr_desc = ms.getcoldesc("FLAG_ROW")
+            fc_desc = ms.getcoldesc("FLAG")
+            ms.close()
+            if os.path.exists(flagvers):
+                os.system(f"rm -fr {flagvers}")
+            from casacore.tables import maketabdesc
+            fr_desc['name'] = "FLAG_ROW"
+            fc_desc['name'] = "FLAG"
+            td = maketabdesc([dict(name='FLAG_ROW', desc=fr_desc), dict(name='FLAG', desc=fc_desc)])
+            ftab = Owlcat.table(flagvers, tabledesc=td, readonly=False, nrow=nr)
+            ftab.putcol("FLAG_ROW", flag_row)
+            ftab.putcol("FLAG", flag_col)
+            ftab.close()
+            ratio = flag_col.sum() / float(flag_col.size)
+            print(f"Flag version '{options.save}' saved: {ratio:.2%} flagged.")
+            # add to list
+            try:
+                version_list_file = f"{msname}.flagversions/FLAG_VERSION_LIST"
+                version_list = open(version_list_file).readlines()
+                versions = set(line.split(':', 1)[0].strip() for line in  version_list)
+                if options.save in versions:
+                    print(f"Flag version '{options.save}' is present in CASA's FLAG_VERSION_LIST.")
+                else:
+                    with open(version_list_file, "wt") as fo:
```

Comment:
possible unicode issue here. It is perhaps better to explicitly open in binary mode and write utf8-encoded strings? I'm not sure if casa supports unicode strings. A little stomach cramp / gut feel tells me not to even go there...

Reply:
I have no idea. I have ceased trying to understand Python encoding behaviour a long time ago. I just nudge things back and forth until they work. This way seems to work with CASA...